File: 9.20/oc网络请求工具类/oc网络请求工具类/ViewController.m
//
//  ViewController.m
//  oc网络请求工具类
//
//  Created by codygao on 16/9/20.
//  Copyright © 2016年 HM. All rights reserved.
//

#import "ViewController.h"
#import "NetworkTools.h"
@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    NSString *url =  @"http://www.weather.com.cn/data/sk/101010100.html";
    [[NetworkTools sharedtools] requestWithType:GET url:url successful:^(id response) {
        
       
        NSLog(@"%@",response);
       
    } failure:^(NSError *erro) {
       
        NSLog(@"%@",erro);
    }];
    
    
    
}




@end

File: 可选类型/可选类型/ViewController.swift
//
//  ViewController.swift
//  可选类型
//
//  Created by codygao on 16/9/17.
//  Copyright © 2016年 HM. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
       demo()
    }
    func demo(){
        let a: Int? = nil
        let b: Int? = nil
        var c: String? = nil
        //解包
        if a != nil,b != nil{
            let result = a!+b!
            print(result)
        }
        let reslut = (a ?? 0) + (b ?? 0)
        print("===\(reslut)")
        c = "这是可选类型"
        print(c)
        
        
    }


}

